clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -184,6 +184,15 @@ static bool IsSyclMathFunc(unsigned BuiltinID) {
   return true;
 }
 
+static bool isKnownGoodDecl(Decl *D) {
```

Comment:
This function is quite insufficient.  It needs to make sure that this isn't SomeNS::__spirv_ocl_printf, or is a member/static member function, etc.  Also, how is __spirv_ocl_printf mangled?  Do we need to make sure that its declaration has C linkage?

Reply:
It is supposed to be mangled.
I'll extend the check to exclude namespace and member functions.

Reply:
Done.

Reply:
Is it? @AlexeySachkov said that it won't be mangled.

Reply:
> Also, how is __spirv_ocl_printf mangled?

I think it will be mangled at LLVM IR level, but here, in Sema we have unmangled names - at least this is my understanding of how clang works

Reply:
Well, unless the function has C linkage, it will be mangled. In Alexey's patch it does not have C linkage.
https://github.com/intel/llvm/pull/835/files/f91daffcaad3479ff4201dee3efbc430ad8548b8#diff-f352938c75fc78e342c9d81a9914a21dR31

SPIR-V Translator documentation also requires all functions to be Itanium-mangled:
https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/master/docs/SPIRVRepresentationInLLVM.rst#spir-v-extended-instructions-mapped-to-llvm-function-calls

Reply:
Then you also have to make sure that this doesn't have C linkage.

Reply:
> Then you also have to make sure that this doesn't have C linkage.

The latest revision of the patch has a check for CXX linkage.

Reply:
And there is no definition for this function - only a declaration.

Reply:
Done.